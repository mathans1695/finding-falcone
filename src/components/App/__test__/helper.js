export const mockHandleResult = (selectedPlanets, selectedVehicles) => undefined;

export const selectedPlanets = ['Donlon', 'Enchai', 'Sapir', 'Pingasor'];
export const selectedVehicles = ['Space pod', 'Space rocket', 'Space ship', 'Space ship'];