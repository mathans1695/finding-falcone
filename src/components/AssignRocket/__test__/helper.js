import _ from 'lodash';

const vehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2, isPossible: false, showAlways: false},
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4, isPossible: false, showAlways: false },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5, isPossible: false, showAlways: false },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10, isPossible: false, showAlways: false }
];
const id = '111111';

const generateVehicles = (planetDistance, selected) => {
	let vehiclesObj = {
		id: id,
		planetDistance: planetDistance,
		vehicles: _.cloneDeep(vehicles),
		previousSelected: ''
	}
	
	vehiclesObj.vehicles.forEach(vehicle => {
		if(vehicle.max_distance >= planetDistance && vehicle.total_no > 0) {
			vehicle.isPossible = true;
		}
	});
	
	if(selected) {
		vehiclesObj.vehicles.forEach(vehicle => {
			if(vehicle.isPossible && vehicle.name === selected){
				vehicle.showAlways = true;
				vehicle.total_no -= 1;
				vehiclesObj.previousSelected = selected;
			}
		})
	}
	
	return vehiclesObj
}

export const initialVehicles = generateVehicles(200);
export const selectVehicles = generateVehicles(200, 'Space pod');
export const disableParticularVehicle = generateVehicles(400, 'Space pod');

export const event = {
	target: {
		value: 'Space rocket',
		getAttribute: (attr) => {
			if(attr === 'data-id') return '111111';
			if(attr === 'data-speed') return 4;
		}
	}
}

export const mockHandleVehicleUpdation = jest.fn((id, rocket, speed, distance) => undefined);