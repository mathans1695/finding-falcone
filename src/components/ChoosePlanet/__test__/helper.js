const planets = [
	{name: "Donlon", distance: 100},
	{name: "Enchai", distance: 200},
	{name: "Jebing", distance: 300},
	{name: "Sapir", distance: 400},
	{name: "Lerbin", distance: 500},
	{name: "Pingasor", distance: 600}
]
const id = "111111";

export const initialPlanets = {
	curPlanet: "Choose Planet",
	id: id,
	planets: planets,
	previousSelected: []
}

export const changePlanets = {
	curPlanet: "Donlon",
	id: id,
	planets: planets,
	previousSelected: [{name: "Donlon", distance: 100}]
}

export const mockUpdateListOfPlanets = jest.fn((id, value, distance) => undefined);
export const mockUpdateListOfVehicles = jest.fn((id, distance) => undefined);

export const event = {
	target: {
		parentElement: { id },
		options: [
			{ selected: false },
			{ selected: false },
			{ selected: true, getAttribute: () => 200 },
			{ selected: false },
			{ selected: false },
			{ selected: false },
			{ selected: false }
		],
		value: 'Enchai'
	}
}