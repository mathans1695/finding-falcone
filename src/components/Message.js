import React, { Component } from 'react';
import '../styles/Message.css';

class Message extends Component {
	render() {
		return (
			<div className='Message'>
				<span className='Message__msg'>
					Selected Rocket is not available
				</span>
			</div>
		)
	}
}

export default Message;