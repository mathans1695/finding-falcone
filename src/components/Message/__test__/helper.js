export const resetMsg = 'Reset successful';
export const notAvailableMsg = 'Selected rocket is not available';