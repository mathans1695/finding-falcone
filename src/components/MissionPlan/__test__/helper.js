export const globalVehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2 },
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4 },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5 },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10 }
];

export const indivVehicles = [
	{name: "Space pod", total_no: 2, max_distance: 200, speed: 2, isPossible: false, showAlways: false},
	{name: "Space rocket", total_no: 1, max_distance: 300, speed: 4, isPossible: false, showAlways: false },
	{name: "Space shuttle", total_no: 1, max_distance: 400, speed: 5, isPossible: false, showAlways: false },
	{name: "Space ship", total_no: 2, max_distance: 600, speed: 10, isPossible: false, showAlways: false }
];

export const getIndex = (vehicles, name) => vehicles.findIndex(vehicle => vehicle.name === name);