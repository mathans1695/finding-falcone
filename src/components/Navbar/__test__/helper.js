export const mockReset = jest.fn();