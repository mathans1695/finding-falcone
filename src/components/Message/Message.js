import React from 'react';
import './Message.css';

const Message = (props) => {
	return (
		<div className='Message'>
			<span className='Message__msg'>
				{props.msg}
			</span>
		</div>
	)
}

export default Message;