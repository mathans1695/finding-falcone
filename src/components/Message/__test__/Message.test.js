import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Message from '../Message';
import { resetMsg, notAvailableMsg } from './helper';

it('Renders reset message correctly', () => {
	const wrapper = shallow(<Message msg={resetMsg} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})

it('Renders not availble message correctly', () => {
	const wrapper = shallow(<Message msg={notAvailableMsg} />);
	
	expect(toJson(wrapper)).toMatchSnapshot();
})