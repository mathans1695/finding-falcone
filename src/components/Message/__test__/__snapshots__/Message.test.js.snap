// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders not availble message correctly 1`] = `
<div
  className="Message"
>
  <span
    className="Message__msg"
  >
    Selected rocket is not available
  </span>
</div>
`;

exports[`Renders reset message correctly 1`] = `
<div
  className="Message"
>
  <span
    className="Message__msg"
  >
    Reset successful
  </span>
</div>
`;
